src/utils/LoggerUtils.ts

```diff
@@ -7,610 +7,1045 @@ import {
-    ChatThreadId: event.threadId,
-    ChatMessageId: event.id
-  });
-};
-
-export const logConvertDeletedMessageEvent = (event: ChatMessageDeletedEvent): void => {
-  Logger.logEvent(LogLevel.INFO, {
-    Event: LogEvent.ACS_ADAPTER_CONVERT_DELETED_MESSAGE,
-    Description: 'ACS Adapter: convert edited deleted message event',
-    CustomProperties: event,
-    MessageSender: (event.sender as CommunicationUserIdentifier).communicationUserId,
-    TimeStamp: event.deletedOn.toISOString(),
-    ChatThreadId: event.threadId,
-    ChatMessageId: event.id
-  });
-};
-
-export const logConvertFetchedMessageFailed = (
-  eventMessage: ChatEventMessage,
-  errorMessage: string,
-  exception: Error
-): void => {
-  Logger.logEvent(LogLevel.ERROR, {
-    Event: LogEvent.ACS_ADAPTER_INGRESS_FAILED,
-    Description: errorMessage,
-    ACSRequesterUserId: eventMessage.currentUserId,
-    MessageSender: (eventMessage.sender as CommunicationUserIdentifier).communicationUserId,
-    TimeStamp: eventMessage.createdOn.toISOString(),
-    ChatThreadId: eventMessage.threadId,
-    ChatMessageId: eventMessage.messageId,
-    ExceptionDetails: exception
-  });
-};
-
-export const logEditEventFailedMetadataParsing = (
-  event: ChatMessageEditedEvent,
-  exception: Error,
-  getState: GetStateFunction<ACSAdapterState>
-): void => {
-  Logger.logEvent(LogLevel.ERROR, {
-    Event: LogEvent.ACS_ADAPTER_CONVERT_HISTORY,
-    Description: 'Failed to parse ChatMessage metadata',
-    ACSRequesterUserId: getState(StateKey.UserId),
-    MessageSender: (event.sender as CommunicationUserIdentifier).communicationUserId,
-    TimeStamp: event.editedOn.toISOString(),
-    ChatThreadId: event.threadId,
-    ChatMessageId: event.id,
-    ExceptionDetails: exception
-  });
-};
-
-export const logFileManagerDownloadFileFailed = (exception: Error): void => {
-  Logger.logEvent(LogLevel.ERROR, {
-    Event: LogEvent.FILEMANAGER_DOWNLOAD_FILE_FAILED,
-    Description: `Downloading attachment for message failed.`,
-    ExceptionDetails: exception
-  });
-};
-
-export const logSDKStartInitError = (
-  threadId: string,
-  requesterUserId: string,
-  description: string,
-  exception?: Error
-): void => {
-  Logger.logEvent(LogLevel.ERROR, {
-    Event: LogEvent.ACS_SDK_START_INIT_ERROR,
-    ChatThreadId: threadId,
-    ACSRequesterUserId: requesterUserId,
-    Description: description,
-    ExceptionDetails: exception
-  });
-};
-
-export const logSDKStartInit = (threadId: string, requesterUserId: string): void => {
-  Logger.logEvent(LogLevel.INFO, {
-    Event: LogEvent.ACS_SDK_START_INIT,
-    ChatThreadId: threadId,
-    ACSRequesterUserId: requesterUserId,
-    Description: `ACS Adapter: ACS Adapter start init.`
-  });
-};
-
-export const logSDKJoinThreadError = (): void => {
-  Logger.logEvent(LogLevel.ERROR, {
-    Event: LogEvent.ACS_SDK_JOINTHREAD_ERROR,
-    Description: `ACS Adapter: failed to join the thread.`
-  });
-};
+export class LoggerUtils {
```

Comment:
I'm not clear why we want a class for this; is there a state that is maintained between objects?
I only see static methods replacing the function definitions.
Would it be cleaner to keep the log methods as pure functions?

Reply:
The main reason is that using a class eliminates the need to import each function individually, which helps avoid having lots imports in some files (such as subscribeNewMessageAndThreadUpdate).

Reply:
Also, it's a little bit clearer than using `import * as LoggerUtils from..."